extensions/basic_auth/plugin.cc

```diff
@@ -244,30 +256,35 @@ bool PluginRootContext::configure(size_t configuration_size) {
 }
 
 FilterHeadersStatus PluginRootContext::check() {
-  auto request_path_header = getRequestHeader(":path");
-  std::string_view request_path = request_path_header->view();
   auto method = getRequestHeader(":method")->toString();
   auto method_iter = basic_auth_configuration_.find(method);
   // First we check if the request method is present in our container
   if (method_iter != basic_auth_configuration_.end()) {
+    auto request_host_header = getRequestHeader(":authority");
+    std::string_view request_host = request_host_header->view();
+    auto request_path_header = getRequestHeader(":path");
+    std::string_view request_path = request_path_header->view();
     // We iterate through our vector of struct in order to find if the
     // request_path according to given match pattern, is part of the plugin's
     // configuration data. If that's the case we check the credentials
     FilterHeadersStatus header_status = FilterHeadersStatus::Continue;
     auto authorization_header = getRequestHeader("authorization");
     std::string_view authorization = authorization_header->view();
-    for (auto& rules : basic_auth_configuration_[method]) {
-      if (rules.pattern == MATCH_TYPE::Prefix) {
-        if (absl::StartsWith(request_path, rules.request_path)) {
-          header_status = credentialsCheck(rules, authorization);
+    for (auto& rule : basic_auth_configuration_[method]) {
+      if (!rule.host.empty() && rule.host != request_host) {
```

Comment:
I am wondering if matching full host:port is expected here. The filter is already called under the context of a listener so why do we need to specify port again ?

Reply:
Stripped port before matching. At Istio 1.10, port will be stripped by default at domain matching.